library/std/src/collections/hash/map.rs

```diff
@@ -2877,13 +2876,24 @@ impl RandomState {
         // iteration order allows a form of DOS attack. To counter that we
         // increment one of the seeds on every RandomState creation, giving
         // every corresponding HashMap a different iteration order.
-        thread_local!(static KEYS: Cell<(u64, u64)> = {
-            Cell::new(sys::hashmap_random_keys())
+        thread_local!(static KEYS: Cell<[u64; 2]> = {
+            let mut buf = [0u8; 16];
+            // Use a constant seed on `wasm32-unknown-unknown` and Hermit targets.
+            // FIXME: remove the Hermit part after its support will be added to `getrandom`
+            // FIXME: replace the WASM part with `cfg(target = "..")`:
+            // https://github.com/rust-lang/rfcs/pull/2991
+            #[cfg(not(any(
+                all(target_arch = "wasm32", target_vendor = "unknown", target_os = "unknown"),
+                all(target_arch = "aarch64", target_os = "hermit"),
+            )))]
+            getrandom::getrandom(&mut buf).expect("failed to get system entropy");
```

Comment:
It might be good to convert the error to an `std::io::Error` first before panicking on it. That way, it uses the Debug implementation of `std`'s Error and we don't pull in any of `getrandom`s display/debug code.